### PR TITLE
D4d

### LINKED ISSUE #7: Cannot enter $15.99 as a value in the price of the item
Currently, only whole numbers such as $15 are allowed.  Investigate why that is the case and what we should do to allow decimals. 

Note that the currency values should be stored using either 'decimal' type or 'integer' type.  We shouldn't use float to store currency value as the calculation using float won't be accurate because of hardware limitation.

### PR DESCRIPTION
**Pull request by Digital4design**
fixes #7 and #10 and #15

1.Ad social links to the user profile page
2.Fix the aspect ratio issue for some issues when viewing the search page on mobile devices
3.Cannot enter 15.99 as a value in the price of the item #7

### CODE REVIEW

**Comment by thihamin** [comment]
when I ran the branch using `quasar dev`, there are some warnings related to formatting.  If they are not difficult to fix, could we fix them?

```
 WARNING  Compiled with 4 warnings                                                   11:52:38 AM

Module Warning (from ./node_modules/eslint-loader/dist/cjs.js):

/Users/thiha/syr/syr-marketplace/src/components/ProfileCard.vue
  122:1  warning  Expected indentation of 12 spaces but found 11 spaces  vue/html-indent

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.


Module Warning (from ./node_modules/eslint-loader/dist/cjs.js):

/Users/thiha/syr/syr-marketplace/src/pages/Asset.vue
  156:1   warning  Expected indentation of 12 spaces but found 10 spaces                        vue/html-indent
  159:1   warning  Expected indentation of 16 spaces but found 15 spaces                        vue/html-indent
  160:1   warning  Expected indentation of 16 spaces but found 15 spaces                        vue/html-indent
  161:1   warning  Expected indentation of 16 spaces but found 15 spaces                        vue/html-indent
  161:16  warning  Attribute "v-model" should go before "@change"                               vue/attributes-order
  162:1   warning  Expected indentation of 16 spaces but found 15 spaces                        vue/html-indent
  162:16  warning  Attribute "required" should go before "@change"                              vue/attributes-order
  162:24  warning  Expected 1 line break before closing bracket, but no line breaks found       vue/html-closing-bracket-newline
  162:24  warning  Expected no space before '>', but found                                      vue/html-closing-bracket-spacing
  163:1   warning  Expected indentation of 16 spaces but found 14 spaces                        vue/html-indent
  164:1   warning  Expected indentation of 18 spaces but found 14 spaces                        vue/html-indent
  165:1   warning  Expected indentation of 18 spaces but found 14 spaces                        vue/html-indent
  165:15  warning  Unexpected 'v-bind' before ':'                                               vue/v-bind-style
  166:1   warning  Expected indentation of 18 spaces but found 14 spaces                        vue/html-indent
  166:15  warning  Unexpected 'v-bind' before ':'                                               vue/v-bind-style
  166:41  warning  Expected 1 line break before closing bracket, but no line breaks found       vue/html-closing-bracket-newline
  167:1   warning  Expected indentation of 18 spaces but found 16 spaces                        vue/html-indent
  168:1   warning  Expected indentation of 16 spaces but found 14 spaces                        vue/html-indent
  169:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  170:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  170:23  warning  Expected 1 line break after opening tag (`<p>`), but no line breaks found    vue/singleline-html-element-content-newline
  170:26  warning  Unexpected usage of 'this'                                                   vue/this-in-template
  170:62  warning  Expected 1 line break before closing tag (`</p>`), but no line breaks found  vue/singleline-html-element-content-newline
  172:1   warning  Expected indentation of 10 spaces but found 12 spaces                        vue/html-indent
  174:1   warning  Expected indentation of 12 spaces but found 10 spaces                        vue/html-indent
  175:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  176:1   warning  Expected indentation of 16 spaces but found 12 spaces                        vue/html-indent
  177:1   warning  Expected indentation of 16 spaces but found 12 spaces                        vue/html-indent
  177:13  warning  Disallow self-closing on HTML void elements (<input/>)                       vue/html-self-closing
  178:1   warning  Expected indentation of 18 spaces but found 13 spaces                        vue/html-indent
  179:1   warning  Expected indentation of 18 spaces but found 13 spaces                        vue/html-indent
  179:14  warning  Attribute "id" should go before "type"                                       vue/attributes-order
  180:1   warning  Expected indentation of 18 spaces but found 13 spaces                        vue/html-indent
  181:1   warning  Expected indentation of 18 spaces but found 13 spaces                        vue/html-indent
  182:1   warning  Expected indentation of 18 spaces but found 13 spaces                        vue/html-indent
  183:1   warning  Expected indentation of 16 spaces but found 13 spaces                        vue/html-indent
  184:1   warning  Expected indentation of 16 spaces but found 13 spaces                        vue/html-indent
  184:14  warning  Require self-closing on HTML elements (<span>)                               vue/html-self-closing
  185:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  186:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  187:1   warning  Expected indentation of 16 spaces but found 12 spaces                        vue/html-indent
  188:1   warning  Expected indentation of 18 spaces but found 12 spaces                        vue/html-indent
  189:1   warning  Expected indentation of 18 spaces but found 12 spaces                        vue/html-indent
  190:1   warning  Expected indentation of 18 spaces but found 12 spaces                        vue/html-indent
  191:1   warning  Expected indentation of 18 spaces but found 12 spaces                        vue/html-indent
  192:1   warning  Expected indentation of 18 spaces but found 12 spaces                        vue/html-indent
  193:1   warning  Expected indentation of 18 spaces but found 12 spaces                        vue/html-indent
  193:39  warning  Expected 1 line break before closing bracket, but no line breaks found       vue/html-closing-bracket-newline
  193:40  warning  Require self-closing on SVG elements (<path>)                                vue/html-self-closing
  194:1   warning  Expected indentation of 16 spaces but found 12 spaces                        vue/html-indent
  195:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  196:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  207:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  208:1   warning  Expected indentation of 16 spaces but found 14 spaces                        vue/html-indent
  209:1   warning  Expected indentation of 18 spaces but found 16 spaces                        vue/html-indent
  210:1   warning  Expected indentation of 18 spaces but found 14 spaces                        vue/html-indent
  210:15  warning  Disallow self-closing on HTML void elements (<input/>)                       vue/html-self-closing
  211:1   warning  Expected indentation of 20 spaces but found 13 spaces                        vue/html-indent
  212:1   warning  Expected indentation of 20 spaces but found 13 spaces                        vue/html-indent
  212:14  warning  Attribute "id" should go before "type"                                       vue/attributes-order
  213:1   warning  Expected indentation of 20 spaces but found 13 spaces                        vue/html-indent
  214:1   warning  Expected indentation of 20 spaces but found 13 spaces                        vue/html-indent
  215:1   warning  Expected indentation of 20 spaces but found 13 spaces                        vue/html-indent
  216:1   warning  Expected indentation of 18 spaces but found 13 spaces                        vue/html-indent
  217:1   warning  Expected indentation of 18 spaces but found 12 spaces                        vue/html-indent
  217:13  warning  Require self-closing on HTML elements (<span>)                               vue/html-self-closing
  218:1   warning  Expected indentation of 16 spaces but found 12 spaces                        vue/html-indent
  219:1   warning  Expected indentation of 16 spaces but found 12 spaces                        vue/html-indent
  220:1   warning  Expected indentation of 18 spaces but found 14 spaces                        vue/html-indent
  221:1   warning  Expected indentation of 18 spaces but found 12 spaces                        vue/html-indent
  221:13  warning  Disallow self-closing on HTML void elements (<input/>)                       vue/html-self-closing
  222:1   warning  Expected indentation of 20 spaces but found 12 spaces                        vue/html-indent
  223:1   warning  Expected indentation of 20 spaces but found 13 spaces                        vue/html-indent
  223:14  warning  Attribute "id" should go before "type"                                       vue/attributes-order
  224:1   warning  Expected indentation of 20 spaces but found 13 spaces                        vue/html-indent
  225:1   warning  Expected indentation of 20 spaces but found 13 spaces                        vue/html-indent
  226:1   warning  Expected indentation of 20 spaces but found 13 spaces                        vue/html-indent
  227:1   warning  Expected indentation of 18 spaces but found 13 spaces                        vue/html-indent
  228:1   warning  Expected indentation of 18 spaces but found 13 spaces                        vue/html-indent
  228:14  warning  Require self-closing on HTML elements (<span>)                               vue/html-self-closing
  229:1   warning  Expected indentation of 16 spaces but found 12 spaces                        vue/html-indent
  230:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  231:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  232:1   warning  Expected indentation of 16 spaces but found 14 spaces                        vue/html-indent
  233:1   warning  Expected indentation of 18 spaces but found 16 spaces                        vue/html-indent
  234:1   warning  Expected indentation of 20 spaces but found 16 spaces                        vue/html-indent
  235:1   warning  Expected indentation of 20 spaces but found 16 spaces                        vue/html-indent
  236:1   warning  Expected indentation of 20 spaces but found 16 spaces                        vue/html-indent
  237:1   warning  Expected indentation of 20 spaces but found 16 spaces                        vue/html-indent
  238:1   warning  Expected indentation of 20 spaces but found 16 spaces                        vue/html-indent
  239:1   warning  Expected indentation of 20 spaces but found 16 spaces                        vue/html-indent
  240:1   warning  Expected indentation of 20 spaces but found 16 spaces                        vue/html-indent
  240:43  warning  Expected 1 line break before closing bracket, but no line breaks found       vue/html-closing-bracket-newline
  240:44  warning  Require self-closing on SVG elements (<path>)                                vue/html-self-closing
  241:1   warning  Expected indentation of 18 spaces but found 16 spaces                        vue/html-indent
  242:1   warning  Expected indentation of 16 spaces but found 12 spaces                        vue/html-indent
  243:1   warning  Expected indentation of 14 spaces but found 12 spaces                        vue/html-indent
  245:1   warning  Expected indentation of 10 spaces but found 12 spaces                        vue/html-indent
  247:1   warning  Expected indentation of 12 spaces but found 10 spaces                        vue/html-indent
  912:7   warning  Unexpected console statement                                                 no-console

✖ 100 problems (0 errors, 100 warnings)
  0 errors and 98 warnings potentially fixable with the `--fix` option.


Module Warning (from ./node_modules/eslint-loader/dist/cjs.js):

/Users/thiha/syr/syr-marketplace/src/pages/NewAsset.vue
  749:19  warning  Attribute "required" should go before "@keypress"       vue/attributes-order
  750:19  warning  Attribute "bottom-slots" should go before "@keypress"   vue/attributes-order
  861:1   warning  Expected indentation of 14 spaces but found 12 spaces   vue/html-indent
  862:1   warning  Expected indentation of 16 spaces but found 13 spaces   vue/html-indent
  863:1   warning  Expected indentation of 18 spaces but found 13 spaces   vue/html-indent
  864:1   warning  Expected indentation of 18 spaces but found 12 spaces   vue/html-indent
  864:13  warning  Disallow self-closing on HTML void elements (<input/>)  vue/html-self-closing
  865:1   warning  Expected indentation of 20 spaces but found 13 spaces   vue/html-indent
  866:1   warning  Expected indentation of 20 spaces but found 13 spaces   vue/html-indent
  866:14  warning  Attribute "id" should go before "type"                  vue/attributes-order
  867:1   warning  Expected indentation of 20 spaces but found 13 spaces   vue/html-indent
  868:1   warning  Expected indentation of 20 spaces but found 13 spaces   vue/html-indent
  868:14  warning  Attribute "v-model" should go before "value"            vue/attributes-order
  869:1   warning  Expected indentation of 18 spaces but found 13 spaces   vue/html-indent
  870:1   warning  Expected indentation of 18 spaces but found 13 spaces   vue/html-indent
  870:14  warning  Require self-closing on HTML elements (<span>)          vue/html-self-closing
  871:1   warning  Expected indentation of 16 spaces but found 12 spaces   vue/html-indent
  872:1   warning  Expected indentation of 16 spaces but found 12 spaces   vue/html-indent
  882:1   warning  Expected indentation of 16 spaces but found 12 spaces   vue/html-indent
  883:1   warning  Expected indentation of 18 spaces but found 14 spaces   vue/html-indent
  884:1   warning  Expected indentation of 20 spaces but found 16 spaces   vue/html-indent
  885:1   warning  Expected indentation of 20 spaces but found 14 spaces   vue/html-indent
  885:15  warning  Disallow self-closing on HTML void elements (<input/>)  vue/html-self-closing
  886:1   warning  Expected indentation of 22 spaces but found 13 spaces   vue/html-indent
  887:1   warning  Expected indentation of 22 spaces but found 13 spaces   vue/html-indent
  887:14  warning  Attribute "id" should go before "type"                  vue/attributes-order
  888:1   warning  Expected indentation of 22 spaces but found 13 spaces   vue/html-indent
  889:1   warning  Expected indentation of 22 spaces but found 13 spaces   vue/html-indent
  889:14  warning  Attribute "v-model" should go before "value"            vue/attributes-order
  890:1   warning  Expected indentation of 20 spaces but found 13 spaces   vue/html-indent
  891:1   warning  Expected indentation of 20 spaces but found 13 spaces   vue/html-indent
  891:14  warning  Require self-closing on HTML elements (<span>)          vue/html-self-closing
  892:1   warning  Expected indentation of 18 spaces but found 12 spaces   vue/html-indent
  893:1   warning  Expected indentation of 18 spaces but found 12 spaces   vue/html-indent
  894:1   warning  Expected indentation of 20 spaces but found 14 spaces   vue/html-indent
  895:1   warning  Expected indentation of 20 spaces but found 12 spaces   vue/html-indent
  895:13  warning  Disallow self-closing on HTML void elements (<input/>)  vue/html-self-closing
  896:1   warning  Expected indentation of 22 spaces but found 12 spaces   vue/html-indent
  897:1   warning  Expected indentation of 22 spaces but found 13 spaces   vue/html-indent
  897:14  warning  Attribute "id" should go before "type"                  vue/attributes-order
  898:1   warning  Expected indentation of 22 spaces but found 13 spaces   vue/html-indent
  899:1   warning  Expected indentation of 22 spaces but found 13 spaces   vue/html-indent
  899:14  warning  Attribute "v-model" should go before "value"            vue/attributes-order
  900:1   warning  Expected indentation of 20 spaces but found 13 spaces   vue/html-indent
  901:1   warning  Expected indentation of 20 spaces but found 13 spaces   vue/html-indent
  901:14  warning  Require self-closing on HTML elements (<span>)          vue/html-self-closing
  902:1   warning  Expected indentation of 18 spaces but found 12 spaces   vue/html-indent
  903:1   warning  Expected indentation of 16 spaces but found 12 spaces   vue/html-indent
  904:1   warning  Expected indentation of 14 spaces but found 12 spaces   vue/html-indent
  907:1   warning  Expected indentation of 14 spaces but found 12 spaces   vue/html-indent
  908:1   warning  Expected indentation of 16 spaces but found 14 spaces   vue/html-indent
  909:1   warning  Expected indentation of 16 spaces but found 14 spaces   vue/html-indent
  910:1   warning  Expected indentation of 18 spaces but found 14 spaces   vue/html-indent
  910:57  warning  Unexpected 'v-bind' before ':'                          vue/v-bind-style
  910:95  warning  Multiple spaces found before 'v-bind'                   vue/no-multi-spaces
  910:97  warning  Unexpected 'v-bind' before ':'                          vue/v-bind-style
  911:1   warning  Expected indentation of 20 spaces but found 16 spaces   vue/html-indent
  912:1   warning  Expected indentation of 18 spaces but found 14 spaces   vue/html-indent
  913:1   warning  Expected indentation of 16 spaces but found 12 spaces   vue/html-indent
  914:1   warning  Expected indentation of 14 spaces but found 12 spaces   vue/html-indent
  916:1   warning  Expected indentation of 12 spaces but found 11 spaces   vue/html-indent

✖ 61 problems (0 errors, 61 warnings)
  0 errors and 61 warnings potentially fixable with the `--fix` option.


Module Warning (from ./node_modules/eslint-loader/dist/cjs.js):

/Users/thiha/syr/syr-marketplace/src/pages/PublicProfile.vue
  273:1   warning  Expected indentation of 10 spaces but found 8 spaces                          vue/html-indent
  273:66  warning  Expected 1 line break after opening tag (`<h2>`), but no line breaks found    vue/singleline-html-element-content-newline
  273:78  warning  Expected 1 line break before closing tag (`</h2>`), but no line breaks found  vue/singleline-html-element-content-newline

✖ 3 problems (0 errors, 3 warnings)
  0 errors and 3 warnings potentially fixable with the `--fix` option.


You may use special comments to disable some warnings.
Use // eslint-disable-next-line to ignore the next line.
Use /* eslint-disable */ to ignore all warnings in a file.
ℹ ｢wds｣: Project is running at http://0.0.0.0:8080/
ℹ ｢wds｣: webpack output is served from /
ℹ ｢wds｣: 404s will fallback to /index.html

```

File: src/pages/NewAsset.vue [hunk 1]
```diff
@@ -53,6 +53,258 @@ export default {
       newUserImages: [], // to add to user images for reuse
       uploaderFiles: [],
       creatingAsset: false,
+      deliverytypes: [],
```
Comment (justinschuldt) [inline src/pages/NewAsset.vue]: @Digital4design attributes are part of the platform. They are dynamic (saved to the DB), and the UI is dynamic, ie the UI will show the options and handle the saving automatically. Check out the docs here https://stelace.com/docs/assets/custom-attributes.html#custom-attributes

Comment (justinschuldt) [reply src/pages/NewAsset.vue]: This is not your fault that you hardcoded this, as Rob did not give you any advice on how to do this.

File: src/components/ProfileCard.vue [hunk 2]
```diff
@@ -116,18 +116,18 @@
                 :value="selectedUser.firstName"
                 :active="isCurrentUser"
                 :custom-save="updateUserFn('firstname')"
-                :input-label="$t({ id: 'user.firstname_label' })"
```
Comment (justinschuldt) [inline src/components/ProfileCard.vue:119]: @Digital4design This app uses i18n multilanguage support. This bit of code here is takes the language definition from the definition file for the language of the user (if supported). Currently this app supports English and French. We want to keep the i18n tooling intact so we can support many languages in the future.

This bit of code (`$t({ id: 'user.firstname_label' })`) will be replace by this definition: https://github.com/justinschuldt/syr-marketplace/blob/dev/src/i18n/source/user.yaml#L6

File: src/components/ProfileCard.vue [hunk 3]
```diff
@@ -266,6 +267,7 @@ export default {
       date: '',
       newAvatarUrl: '',
       uploadingAvatarSrc: '',
+      labels: { name: 'Name' },
```
Comment (justinschuldt) [inline src/components/ProfileCard.vue]: @Digital4design all copy should be added to the appropriate definition file.
This copy should come from https://github.com/justinschuldt/syr-marketplace/blob/dev/src/i18n/source/user.yaml

File: src/index.template.html [hunk 4]
```diff
@@ -3,10 +3,32 @@
   <head>
     <!-- Injected dynamically by seo mixin -->
     <!-- <title><%= productName %></title> -->
+    <style>
+      @media only screen and (max-width:767px)
+      {
+   .asset-image-container img.asset-image 
+{
```
Comment (justinschuldt) [inline src/index.template.html]: @Digital4design please format your code in the same style as the rest of the code in this repo. This should be indented. The development server should tell you about linting errors. You can check your formatting by running `yarn lint`, and you can automatically fix some formatting errors by running `yarn lint:fix`.

File: src/index.template.html [hunk 5]
```diff
@@ -3,10 +3,32 @@
...
+    display: flex;
+    justify-content: space-between;
+} 
+      }
+    </style>
     <meta charset="utf-8">
     <!-- Injected dynamically by seo mixin -->
-    <!-- <meta name="description" content="<%= productDescription %>"> -->
+    <!-- <meta name="description" content="w<%= productDescription %>"> -->
```
Comment (justinschuldt) [inline src/index.template.html]:
```suggestion
    <!-- <meta name="description" content="<%= productDescription %>"> -->
```
@Digital4design this `w` looks like a typo, please remove.

File: src/pages/Asset.vue [hunk 6]
```diff
@@ -152,9 +152,99 @@
             allow-falsy-save
             input-type="textarea"
           />
-
+          <div class="row">
+          <div class="country col-12 col-sm-7">
```
Comment (justinschuldt) [inline src/pages/Asset.vue]: @Digital4design attribute display and editing is automatic, sorry you did not have good direction when you started this.

Custom attributes should be added to the database via API (https://docs.api.stelace.com/#c66f640c-ad98-47a5-975e-57904566d4fb), then the display is automatic along with the controls to edit https://github.com/justinschuldt/syr-marketplace/blob/dev/src/components/CustomAttributesEditor.vue.
I will add the custom attributes to the DB. Sorry Rob failed to provide direction for this.

File: src/pages/Asset.vue [hunk 7]
```diff
@@ -152,9 +152,99 @@
...
+             <span class="checkmark"></span>
+            </label>
+            <label v-else class="">
+            <svg
+            aria-hidden="true"
+            role="presentation"
+            focusable="false"
+            viewBox="0 0 24 24"
+            class="q-mr-sm q-icon notranslate text-secondary"
+            style="font-size: 1.5rem;"><path d="M14.12,10H19V8.2H15.38L13.38,4.87C13.08,4.37 12.54,4.03 11.92,4.03C11.74,4.03 11.58,4.06 11.42,4.11L6,5.8V11H7.8V7.33L9.91,6.67L6,22H7.8L10.67,13.89L13,17V22H14.8V15.59L12.31,11.05L13.04,8.18M14,3.8C15,3.8 15.8,3 15.8,2C15.8,1 15,0.2 14,0.2C13,0.2 12.2,1 12.2,2C12.2,3 13,3.8 14,3.8Z"></path></svg>
```
Comment (justinschuldt) [inline src/pages/Asset.vue]: @Digital4design please use icons in the same way that other icons are used in this file, even though it's hacky:

- import: https://github.com/justinschuldt/syr-marketplace/blob/dev/src/pages/Asset.vue#L333
- add to scope: https://github.com/justinschuldt/syr-marketplace/blob/dev/src/pages/Asset.vue#L500
- translate attribute property to component in scope: https://github.com/justinschuldt/syr-marketplace/blob/dev/src/pages/Asset.vue#L406

File: src/pages/Asset.vue [hunk 8]
```diff
@@ -359,6 +449,253 @@ export default {
       assetRatingsLoaded: false,
       isEditingImages: false,
       isPlaceSearchEnabled,
+      selected: '',
+      optionssel: [
```
Comment (justinschuldt) [inline src/pages/Asset.vue]: @Digital4design Typically if you have constants that are used in several files, you should not duplicate the constants in multiple places. Check out the [DRY principle of software engineering](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) if you're interested.
Constants could be added to one file, then imported wherever you need them. That way when someone updates the list, they only need to do it in one spot.

I will put this data in a custom attribute in the DB, so no need to actually move this data, you can just remove it.

File: src/pages/PublicProfile.vue [hunk 9]
```diff
@@ -240,7 +265,94 @@ export default {
...
+          <!-- reuse generic asset.description_label -->
+
+          <AppSwitchableEditor
+            tag="p"
+            class="text-body1 q-ma-lg text-justify"
+            :class="{ 'tktk': !isCurrentUser }"
+            :value="selectedUser.metadata.tiktok"
+            :active="isCurrentUser"
+            :custom-save="updateSocUserFn('tiktok')"
+            :input-label="social.twitter"
```
Comment (justinschuldt) [inline src/pages/PublicProfile.vue]: Should this be `social.tiktok`?

File: src/pages/PublicProfile.vue [hunk 10]
```diff
@@ -316,6 +428,37 @@ export default {
 </template>
 
 <style lang="stylus" scoped>
+.instagn {
+    background: url(https://d328vk3fbp0jh4.cloudfront.net/eyJidWNrZXQiOiJzeXItbWFya2V0LWltYWdlcy1wcm9kIiwia2V5IjoiZGV2L2ltYWdlcy80OWMwNjQ4NWFiYzE2NGUwOWNhZTdjYzMxOWMzMzc4MC1pbnN0YWdyYW0ucG5nIiwiZWRpdHMiOnsid2VicCI6dHJ1ZSwicmVzaXplIjp7IndpZHRoIjoxMjAwLCJoZWlnaHQiOjgwMCwid2l0aG91dEVubGFyZ2VtZW50Ijp0cnVlfX19);
```
Comment (justinschuldt) [inline src/pages/PublicProfile.vue]: @Digital4design I don't think making an HTTP request for each of these is a good idea since they could be included in the app's source via an icon, or if these are not available as Quasar icons, an SVG would be better. Please replace these with Quasar icons or SVGs.

File: src/pages/PublicProfile.vue [hunk 11]
```diff
@@ -240,7 +265,94 @@ export default {
             input-type="textarea"
           />
         </section>
+        <QSeparator class="q-mt-xl" />
+        <section
+          v-show="!(selectedUser.id && !isCurrentUser && !selectedUser.metadata)"
+          class="q-px-sm"
+        >
+        <h2 class="text-h4 text-weight-medium stl-content-entry">Social links</h2>
```
Comment (justinschuldt) [inline src/pages/PublicProfile.vue]: @Digital4design All copy should be in language definition files https://github.com/justinschuldt/syr-marketplace/tree/dev/src/i18n/source
Please add the copy for this to the `user.yaml` file under a new key named `social_links_label`, then you can update this to use the value with: 
```suggestion
          <AppContent
            v-if="selectedUser.metadata"
            tag="h2"
            class="text-h4 text-weight-medium"
            entry="user"
            field="social_links_label"
          />
```
Note, it is much better to match the existing coding style for showing a label, like in this case to use `<AppContent ... />`, rather than using a plain `<h2>` and adding your own styles.

File: src/pages/PublicProfile.vue [hunk 12]
```diff
@@ -135,6 +142,24 @@ export default {
...
+      obj[fieldName] = value
+      await this.$store.dispatch('updateUser', {
+        userId: this.selectedUser.id,
+        attrs: {
+          [meta]: obj
+        }
+      })
+      this.notifySuccess('notification.saved')
+
+      // Hack to erase the content in locations AppSwitchableEditor (c.f. `computed.locationValue`)
```
Comment (justinschuldt) [inline src/pages/PublicProfile.vue]: @Digital4design why is this needed? I would think this function would only be called when editing the social links.

File: src/pages/PublicProfile.vue [hunk 13]
```diff
@@ -37,6 +37,7 @@ export default {
       isPlaceSearchEnabled,
       userRatingsByTransaction: [],
       userRatingsLoaded: false,
+      social: { instagram: 'Instagram Link', facebook: 'Facebook Link', twitter: 'Tiktok Link', website: 'Clubhouse Link' },
```
Comment (justinschuldt) [inline src/pages/PublicProfile.vue]: @Digital4design Please put these in the language definition file: https://github.com/justinschuldt/syr-marketplace/blob/dev/src/i18n/source/user.yaml

Also, why is it `twitter: 'Tiktok Link', website: 'Clubhouse Link'`? Can you please update these to be consistent, if we telling the user to input tiktok and clubhouse links, we should name the internal properties to accurately reflect the data that is being captured. Please update the metadata property names and the css classes to match the labels.

**Review by justinschuldt — CHANGES_REQUESTED** [review]
Nice work, thanks for this!

Please remove all the code for the attributes (delivery and cuisine), I will add those to the DB.

I left some comments for changes, and just general feedback. Feel free to respond to comments if you'd like to discuss. Let me know when you've pushed updates and this is ready for me to re-review.

Thanks!

File: src/pages/PublicProfile.vue (hunk 12 above)
Comment (Digital4design) [reply src/pages/PublicProfile.vue]: yes it would be call after saving the social link fields

File: src/components/ProfileCard.vue (hunk 2 above)
Comment (Digital4design) [reply src/components/ProfileCard.vue:119]: done replaced it with $t({ id: 'user.name_label' })

File: src/components/ProfileCard.vue (hunk 3 above)
Comment (Digital4design) [reply src/components/ProfileCard.vue]: i had removed it and fetch the labels from user.yaml file

File: src/index.template.html (hunk 4 above)
Comment (Digital4design) [reply src/index.template.html]: done, thanks for your suggestion

File: src/index.template.html (hunk 5 above)
Comment (Digital4design) [reply src/index.template.html]: you want to remove just w or full line?

File: src/pages/Asset.vue (hunk 8 above)
Comment (Digital4design) [reply src/pages/Asset.vue]: change selected to selectedcountry == done

File: src/pages/PublicProfile.vue (hunk 9 above)
Comment (Digital4design) [reply src/pages/PublicProfile.vue]: removed , and use like this
$t({ id: 'user.social_links_label.tiktoklabel' })
added social lables in user.yaml and i used them to show lables

File: src/pages/PublicProfile.vue (hunk 10 above)
Comment (Digital4design) [reply src/pages/PublicProfile.vue]: removed , i have add the icons in public folder and fetch from like this 
background: url(/instagram.png);
working successfully

File: src/pages/PublicProfile.vue (hunk 11 above)
Comment (Digital4design) [reply src/pages/PublicProfile.vue]: done as per you described

File: src/pages/PublicProfile.vue (hunk 13 above)
Comment (Digital4design) [reply src/pages/PublicProfile.vue]: removed form file and add these social labels in teh user.yaml and used it to show labels

File: src/pages/NewAsset.vue (hunk 1 above)
Comment (Digital4design) [reply src/pages/NewAsset.vue]: warnings are removed , you can test it 
and i had check this new two fields are added in Database and coming at page publish your add , i guess these are custom attributes so these are managed by backend developer or i have to manage it , please explain
image

**Review by justinschuldt — APPROVED** [review]
I changed a few things, made the social links clickable.
Going to merge now! Thanks!

File: src/pages/PublicProfile.vue (hunk 9 above)
Comment (justinschuldt) [reply src/pages/PublicProfile.vue]: Nice!

File: src/index.template.html (hunk 5 above)
Comment (justinschuldt) [reply src/index.template.html]: just the `w` that was added. I removed it.

File: src/pages/NewAsset.vue (hunk 1 above)
Comment (justinschuldt) [reply src/pages/NewAsset.vue]: I added this to the DB, so we don't need it in the frontend.

File: src/pages/PublicProfile.vue (hunk 12 above)
Comment (justinschuldt) [reply src/pages/PublicProfile.vue]: I removed this, was not needed.
